deployment/cdk/opensearch-service-migration/lib/network-stack.ts

```diff
@@ -9,13 +9,38 @@ import {StackPropsExt} from "./stack-composer";
 import {StringParameter} from "aws-cdk-lib/aws-ssm";
 
 export interface NetworkStackProps extends StackPropsExt {
-    readonly vpcId?: string
-    readonly availabilityZoneCount?: number
+    readonly vpcId?: string,
+    readonly availabilityZoneCount?: number,
+    readonly targetClusterEndpoint?: string
 }
 
 export class NetworkStack extends Stack {
     public readonly vpc: IVpc;
 
+    // Validate a proper url string is provided and return an url string which contains a protocol, host name, and port.
+    // If a port is not provided, the default protocol port (e.g. 443, 80) will be explicitly added
+    static validateAndReturnFormattedHttpURL(urlString: string) {
+        // URL will throw error if the urlString is invalid
+        let url = new URL(urlString);
+        if (url.protocol !== "http:" && url.protocol !== "https:") {
+            throw new Error(`Invalid url protocol for target endpoint: ${urlString} was expecting 'http' or 'https'`)
+        }
+        // URLs that contain the default protocol port (e.g. 443, 80) will not show in the URL toString()
+        let formattedUrlString = url.toString()
+        if (formattedUrlString.endsWith("/")) {
```

Comment:
If a users specifies http://host/directory/ you'll end up keeping 'directory'.
I'd parse the URL as you do, then check that the path/query string is empty (throw an error if it isn't) and then reconstruct the URL by concatenating the protocol, host, port (with a default if it wasn't included).  Formatting a string and parsing it is tedious and error prone.

Reply:
Have adjusted to throw error for paths that are provided. Its worth pointing out here that by default the URL class here will always have "/" as the path unless an actual one is provided (which we now throw an error for) so we remove it for our structure